routes/movies.js

```diff
@@ -43,83 +37,114 @@ router.post('/', async (request, response, next) => {
-        return new array with both movies
-      */
-      if (data.length < 2) {
-        const error = { message: 'Not enough movies in the database.' }
-        return next({ status: 404, message: error.message })
-      }
+router.get('/voting/:id', async (request, response, next) => {
+  const { id } = request.params
 
-      const firstMovieIndex = Math.floor(Math.random() * data.length)
-      let secondMovieIndex = firstMovieIndex
-      let gotSecondMovie = false
+  try {
+    const user = await User.findById(id)
+    if (!user) {
+      console.error('User does not exist')
+      return next({ status: 404, message: 'User does not exist' })
+    }
 
-      do {
-        secondMovieIndex = Math.floor(Math.random() * data.length)
-        if (firstMovieIndex !== secondMovieIndex) {
-          gotSecondMovie = true
-        }
-      } while (!gotSecondMovie)
+    const movies = await Movie.find()
+    if (!movies) {
+      console.error('Error receiving movies from database')
+      return next({
+        status: 404,
+        message: 'Error receiving movies from database',
+      })
+    }
 
-      const dataVoting = [data[firstMovieIndex], data[secondMovieIndex]]
+    if (movies.length - user.unwatchedMovies.length < 2) {
+      const error = { message: 'Not enough movies in the database.' }
+      console.error(error.message)
+      next({ status: 400, message: error.message || 'No documents found' })
+    }
 
-      response.status(200).json(dataVoting)
-    })
-    .catch(error =>
-      next({ status: 404, message: error.message || 'No documents found' })
-    )
+    let firstMovieIndex
+    let gotFirstMovie = false
+    do {
+      firstMovieIndex = Math.floor(Math.random() * movies.length)
+      if (!user.unwatchedMovies.includes(movies[firstMovieIndex]._id)) {
+        gotFirstMovie = true
+      }
+    } while (!gotFirstMovie)
+
+    let secondMovieIndex = firstMovieIndex
+    let gotSecondMovie = false
+
+    do {
+      secondMovieIndex = Math.floor(Math.random() * movies.length)
+      if (
+        firstMovieIndex !== secondMovieIndex &&
+        !user.unwatchedMovies.includes(movies[secondMovieIndex]._id)
+      ) {
+        gotSecondMovie = true
+      }
+    } while (!gotSecondMovie)
+
+    const dataVoting = [movies[firstMovieIndex], movies[secondMovieIndex]]
+
+    response.status(200).json(dataVoting)
+  } catch (error) {
+    console.error(error)
+    return next({ status: 500, message: 'Server error' })
+  }
 })
 
-router.get('/:id', (request, response, next) => {
+router.get('/:id', async (request, response, next) => {
   const { id } = request.params
 
-  Movie.findById(id)
-    .then(data => {
-      if (!data) {
-        throw new Error('The movie does not exist!')
-      }
-      response.status(200).json(data)
-    })
-    .catch(error =>
-      next({ status: 404, message: error.message || 'Document not found' })
-    )
+  try {
+    const movie = await Movie.findById(id)
```

Comment:
I am not sure if this is required in mongoose, but I would check if `id` (and other ids in this file) are of type `ObjectId` before fetching from the database.